### ISSUE TITLE
click append sprites to bones not working?

### ISSUE THREAD

**Issue by j-mation** [ndee85/coa_tools#32]
In edit armature mode you are meant to be able to quickly parent sprites to bones.
-press "F" for pie menu and choose edit armature
-After selecting a bone and holding done ALt the highlight selection works for sprites but clicking a sprite has no effect on parenting the sprite to the bone.

Am I missing something or is it broken?

**Comment by j-mation** [comment]
Found the issue! In The Blender user preferences you cannot have "Emulate 3 Button Mouse" active. Unfortunately this is the mode that makes blender really fast to navigate.

**Comment by rosshadden** [comment]
I have the same issue, but my "Emulate 3 button mouse" setting was not enabled.  I tried enabling it anyway, but this did not solve the issue.

All of my settings are default.

**Comment by rosshadden** [comment]
Since this shortcut doesn't work, what's the long-form way of doing this?  I think I remember one of the videos alluding to it, but I don't know what it actually is.

**Comment by ndee85** [comment]
@rosshadden it should work. Do you use a current version? I have made a workaround which disables the 3 mouse button during the armature mode and later resets it again.

**Comment by rosshadden** [comment]
Yeah, I am using the current `master`, as of 340337b.  It is very possible it is just a misunderstanding on my part.  I am not a power user of Blender by any stretch.

With a bone selected, holding `alt` and left-clicking a bone doesn't seem to have an effect, and if I exit the armature editing mode and move a sprite or bone, they are unlinked.

**Comment by ndee85** [comment]
The workflow is as follows. Enter armature mode by using the operator. It will only work if you go  into edit mode by using the Armature Mode Operator. That way you can draw bones in edit mode and you can alt+click on sprites to create a weightmap and bind the sprite to a bone.
If you into the edit mode by using tab. This won't work. You won't be able to draw bones and you won't be able to link sprites with bones.

Alt+click works only in combination with bones and sprites. Not bones and bones.

Hope this makes it clearer. Please tell me if you get it to work. I really need to create a tutorial series for coa tools :)

**Comment by rosshadden** [comment]
I do understand the workflow.  I'm pretty sure something is wrong (in my environment at least).  I'll add some debug code and try to troubleshoot a bit and report back here.

**Comment by ndee85** [comment]
otherwise prepare a file that fails for you and I'll take a look from my side.
Does it work if you disable 3 button mouse?

**Comment by rosshadden** [comment]
No it does not work with 3 button mouse disabled.  Disabled is the default setting, and I had not changed it.  I did try toggling it, but that setting did not make it work.

It's more likely something going on with my system (Arch Linux) than the file itself.  All I am doing for the file is starting from scratch, and importing images from Gimp.  But just in case it is something in that process, I'll import your sample `stip` model in blender and try to do it with that.

**Comment by ndee85** [comment]
Are you using left/right button switch?

**Comment by rosshadden** [comment]
Lol no, all of my settings are default.

The only thing I can think of is that my `alt` key is... [special](https://github.com/rosshadden/dotfiles/blob/482fd72be8459858833b910b1c84852e4ccd1ca2/bin/keys#L14).  When I get a chance I'll play with it, and try just normal `alt`.  Though if that _is_ the problem, it would be the first issue with it that I have noticed in ~3 years :).  I'm just throwing out ideas.

**Comment by ndee85** [comment]
ah... ok.. I remember everytime I setup linux I need to change the alt key. Because Ubuntu at least uses the alt key to move windows.
Please try using alt+rightclick on a edgeloop in editmode and see if a loop gets selected. If so the alt key is occupied by the os for sure.

**Comment by hillac** [comment]
I've noticed that the alt+click to attach sprites to bones does not when in camera view. Being in front-ortho view, however, does work.

**Comment by Larpon** [comment]
With Blender 2.78a I get an icon of what looks like a dropper (color picker tool??) - any suggestions on what to do? The Alt key works e.g. alt+rightclick does an edgeloop in editmode

**Comment by cp38** [comment]
I have a similar issue
I'm on linux mint 18 - blender 2.79
at first I selected bones and pressing ALT in edit mode cannot highligt png to bind (cursor changed to picker).
Then switched from 2d to 3d and now holding alt highlight the png but  leftmouse click wont bind the sprite to the bone.

just a side note that could be added to your installation notes for linux
before creating the zip file for blender or the addon for gimp all .py files needs property->permissions->execution or they wont work

anyway incredibly awesome piece of software
thankyou alot for all the time and effort you put in it !
